### PR TITLE
Updated set command to support tab completion of values

### LINKED ISSUE #870: setting settable attribute with default type None does not change
It appears that when a `settable` attribute is set that a call to `utils.cast` is made and attempts to cast the new `value` to the type of `orig_value`. When the original value is `None` the result of that value is also `None` and, in my case, gets passed to my setter as `None`.

While I understand that this could easily resolved by setting the default value of my attribute to '', it visually looks odd when every other settable is filled besides the one that was not set.

``` text
...
allow_style: Terminal
att: 
continuation_prompt: > 
...
```

Likewise, setting the default value to `'None'` does not seem like a good answer to resolving the issue. 

In this particular case, I think I would prefer if the original value is 'None' then the str that was passed is set

### PR DESCRIPTION
**Pull request by kmvanbrunt**
Closes #870 #871

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@
         * `continuation_prompt`
         * `self_in_py`
         * `prompt`
+    * `self.settable` changed to `self.settables`
```
Comment (tleonhardt) [inline CHANGELOG.md]: We should probably add something under `Enhancements` about the `add_settable()` and `remove_settable()` methods

**Comment by kotfu** [comment]
This all looks pretty good to me. I'll add one suggestion.

If the list of valid values is not known at initialization time (which is when settables would typically get defined), you have to do really awkward stuff in order to validate the new value of a setting when a user changes it. The ``onchange_cb`` doesn't have access to `self` (i.e. the `cmd2` subclass), so my callback doesn't have access to all the data in my app.

If I somehow manage to say the input isn't valid, there is no way for my callback to to say "hey this isn't a valid value, I don't want to allow the user to change it" back to `cmd2`. I think throwing a `ValueError` exception would be ideal here, and `cmd2` could then catch it and not allow the change in the value.

File: docs/features/settings.rst [hunk 2]
```diff
@@ -165,4 +165,4 @@ the :ref:`features/settings:debug` setting. To do so, remove it from the
 
     def __init__(self):
         super().__init__()
-        self.settable.pop('debug')
+        self.settables.pop('debug')
```
Comment (tleonhardt) [inline docs/features/settings.rst]: This should be updated to use the new `self.remove_settable()` method

File: examples/remove_settable.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+A sample application for cmd2 demonstrating how to remove one of the built-in runtime settable parameters.
+"""
+import cmd2
+
+
+class MyApp(cmd2.Cmd):
+
+    def __init__(self):
+        super().__init__()
+        self.settables.pop('debug')
```
Comment (tleonhardt) [inline examples/remove_settable.py]: This should be updated to use the new `self.remove_settable()` method

**Comment by kmvanbrunt** [comment]
@kotfu The `onchange_cb` can have access to `self` if you set it to a bound method of your app.

```Python
onchange_cb = self.val_changed
```

As far as validation, that's the job of `Settable.val_type`. It corresponds to the `type` field in the `argparse.add_argument()` function. Any exception it raises will be sent to `perror()` and the value will not be changed.

**Comment by kotfu** [comment]
Go it. I missed that `val_type` was a callable. You already have implemented my suggestion. This looks good to me.

**Comment by kmvanbrunt** [comment]
@kotfu I update the docstring for `Settable` to be more clear.

**Comment by kotfu** [comment]
hold up before we merge this, I need to fix the settings.rst documentation and update examples/environment.py

File: docs/conf.py [hunk 4]
```diff
@@ -25,9 +25,6 @@
 # Import for custom theme from Read the Docs
 import sphinx_rtd_theme
 
-sys.path.insert(0, os.path.abspath('..'))
```
Comment (tleonhardt) [inline docs/conf.py]: I think you need to delete the import of `os` now that it is unused to make `flake8` happy

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,13 +1,20 @@
-## 0.9.26 (TBD, 2020)
+## 1.0.0-rc1 (TBD, 2020)
```
Comment (tleonhardt) [inline CHANGELOG.md]: Should we maybe go with something like `0.99.0` instead to technically stick within SemVer format?

Comment (kotfu) [reply CHANGELOG.md]: By my interpretation of https://semver.org/#spec-item-9 it is a valid SemVer format.

Comment (tleonhardt) [reply CHANGELOG.md]: Ok cool, I wasn't aware of that.  Sounds good.

**Comment by tleonhardt** [comment]
This looks good so I'm going to merge it

**Pull request merged**